[Bug]: Wurm docker container does not start server

### User story

As a Docker user, I want the Wurm server to start so that I can host the server.

### Game

Wurm

### Linux distro

Ubuntu 22.04

### Command

command: start

### Further information

When starting the ghcr.io/gameservermanagers/gameserver:wurm image the server does not start, either automatically, manually using the /app/wurmserver launcher, or by manually running the execution command.

The issue appears to be due to Java within the container.

I was able to troubleshoot this issue by adding the repository list back into the container and installing the openjdk-8-jre package. After that is present in the container the server will launch without any issues.

I'm opening this issue to link to a pull request for updating the dependency list so that it is included in the container going forward.

### Relevant log output

```shell
Output from ./wurmserver st being run manually:

$ ./wurmserver st
[  OK  ] Starting wurmserver: Test server
[ INFO ] Stopping wurmserver: Test server is already stopped
[  OK  ] Starting wurmserver: Test server
Error! Command did not exit correctly: st

core_exit.sh exiting with code: 2


Output from using the execution command manually in the /data/serverfiles directory:

$ xvfb-run ./WurmServerLauncher ADMINPWD="adminYijhc94Z" EPICSETTINGS="false" EXTERNALPORT="3724" HOMESERVER="true" HOMEKINGDOM="4" LOGINSERVER="true" MAXPLAYERS="200" QUERYPORT="27017" SERVERNAME="LinuxGSM" SERVERPASSWORD="" START="Creative" IP=""
Starting
Logs found: 0
SteamAppId set.
/data/serverfiles
Runtime path: /data/serverfiles/runtime/jre1.8.0_172/lib/amd64/server/libjvm.so
Working Directory: /data/serverfiles
Number of options: 7
Create java VM!
Jul 14, 2023 1:08:33 PM com.wurmonline.server.gui.WurmServerGuiMain main
INFO: WurmServerGuiMain starting
Should start without GUI here!

Execution from starting without any parameters (per the fix_wurm.sh) to generate the Creative/Adventure directories:

$ xvfb-run ./WurmServerLauncher
Starting
Logs found: 0
SteamAppId set.
/data/serverfiles
Runtime path: /data/serverfiles/runtime/jre1.8.0_172/lib/amd64/server/libjvm.so
Working Directory: /data/serverfiles
Number of options: 7
Create java VM!
Jul 14, 2023 1:09:11 PM com.wurmonline.server.gui.WurmServerGuiMain main
INFO: WurmServerGuiMain starting
Graphics Device initialization failed for :  es2, sw
Error initializing QuantumRenderer: no suitable pipeline found
java.lang.RuntimeException: java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found
        at com.sun.javafx.tk.quantum.QuantumRenderer.getInstance(QuantumRenderer.java:280)
        at com.sun.javafx.tk.quantum.QuantumToolkit.init(QuantumToolkit.java:221)
        at com.sun.javafx.tk.Toolkit.getToolkit(Toolkit.java:248)
        at com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:209)
        at com.sun.javafx.application.LauncherImpl.startToolkit(LauncherImpl.java:675)
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:695)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$154(LauncherImpl.java:182)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found
        at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.init(QuantumRenderer.java:94)
        at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:124)
        ... 1 more
```


### Steps to reproduce

1. Download the ghcr.io/gameservermanagers/gameserver:wurm image
2. Run the container
3. The server will not launch

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -52,7 +52,7 @@ queryport="27016"
 
 ## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
 # Parameters are edited in the game server config file.
-startparameters="start=${gamemode} adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryport='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' start='${start}' ip='${ip}'"
+startparameters="start=${gamemode} adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryport='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' ip='${ip}'"
 
 #### LinuxGSM Settings ####
 

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -9,6 +9,8 @@
 #### Game Server Settings ####
 
 ## Predefined Parameters | https://docs.linuxgsm.com/configuration/start-parameters
+# https://www.wurmpedia.com/index.php/Server_administration_(Wurm_Unlimited)
+
 # Game mode (Creative|Adventure)
 gamemode="Creative"
 

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -25,7 +25,6 @@ gamemode="Creative"
 #  4 - Freedom
 homekingdom="4"
 
-
 # Unlocks the admin commands from within the game, that can be used to change
 # the gameplay settings of the server, such as skill gain rate and field growth time.
 adminpassword="ADMINPASSWORD"
@@ -202,10 +201,6 @@ glibc="2.14"
 systemdir="${serverfiles}"
 executabledir="${systemdir}"
 executable="./WurmServerLauncher"
-servercfgdir="${systemdir}"
-servercfg="${selfname}.cfg"
-servercfgdefault="server.cfg"
-servercfgfullpath="${servercfgdir}/${servercfg}"
 
 ## Backup Directory
 backupdir="${lgsmdir}/backup"

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -52,7 +52,7 @@ queryport="27016"
 
 ## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
 # Parameters are edited in the game server config file.
-startparameters="start=${gamemode} adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryprort='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' start='${start}' ip='${ip}'"
+startparameters="start=${gamemode} adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryport='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' start='${start}' ip='${ip}'"
 
 #### LinuxGSM Settings ####
 

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -8,9 +8,47 @@
 
 #### Game Server Settings ####
 
+## Predefined Parameters | https://docs.linuxgsm.com/configuration/start-parameters
+# Game mode (Creative|Adventure)
+gamemode="Creative"
+
+# The kingdom the server belongs to.
+# If you're using the Adventure base, use 1, 2, or 3.
+# If you're using Creative, use 4.
+# Kingdom numbers:
+#  0 - No kingdom
+#  1 - Jen-Kellon
+#  2 - Mol-Rehan
+#  3 - Horde of the Summoned
+#  4 - Freedom
+homekingdom="4"
+
+
+# Unlocks the admin commands from within the game, that can be used to change
+# the gameplay settings of the server, such as skill gain rate and field growth time.
+adminpassword="ADMINPASSWORD"
+
+# If true the server will follow the rules from the Epic servers in Wurm Online.
+# For instance it will use the skill curve.
+epicsettings="false"
+
+# If the server is a home server (belongs to a single kingdom).
+homeserver="true"
+
+# Defaults to true, should only be set to false if the server is intended to be
+# connected with another server that is going to act as a loginserver.
+loginserver="true"
+
+maxplayers="200"
+servername="LinuxGSM"
+serverpassword=""
+ip="0.0.0.0"
+port="3724"
+queryport="27016"
+
 ## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
 # Parameters are edited in the game server config file.
-startparameters="ADMINPWD=\"$ADMINPWD\" EPICSETTINGS=\"$EPICSETTINGS\" EXTERNALPORT=\"$EXTERNALPORT\" HOMESERVER=\"$HOMESERVER\" HOMEKINGDOM=\"$HOMEKINGDOM\" LOGINSERVER=\"$LOGINSERVER\" MAXPLAYERS=\"$MAXPLAYERS\" QUERYPORT=\"$QUERYPORT\" SERVERNAME=\"$SERVERNAME\" SERVERPASSWORD=\"$SERVERPASSWORD\" START=\"$START\" IP=\"$IP\""
+startparameters="start=${gamemode} adminpwd=${adminpassword} epicsettings=${epicsettings} externalport=${port} homeserver=${homeserver} homekingdom=${homekingdom} loginserver=${loginserver} maxplayers=${maxplayers} queryprort=${queryport} servername=${servername} serverpassword=${serverpassword} start=${start} ip=${ip}"
 
 #### LinuxGSM Settings ####
 
@@ -157,13 +195,11 @@ glibc="2.14"
 ## Game Server Directories
 systemdir="${serverfiles}"
 executabledir="${systemdir}"
-preexecutable="xvfb-run"
 executable="./WurmServerLauncher"
 servercfgdir="${systemdir}"
 servercfg="${selfname}.cfg"
 servercfgdefault="server.cfg"
 servercfgfullpath="${servercfgdir}/${servercfg}"
-source "${servercfgfullpath}"
 
 ## Backup Directory
 backupdir="${lgsmdir}/backup"

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -39,6 +39,9 @@ homeserver="true"
 # connected with another server that is going to act as a loginserver.
 loginserver="true"
 
+rmiport="7221"
+rmiregport="7220"
+
 maxplayers="200"
 servername="LinuxGSM"
 serverpassword=""
@@ -48,7 +51,7 @@ queryport="27016"
 
 ## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
 # Parameters are edited in the game server config file.
-startparameters="start=${gamemode} adminpwd=${adminpassword} epicsettings=${epicsettings} externalport=${port} homeserver=${homeserver} homekingdom=${homekingdom} loginserver=${loginserver} maxplayers=${maxplayers} queryprort=${queryport} servername=${servername} serverpassword=${serverpassword} start=${start} ip=${ip}"
+startparameters="start=${gamemode} adminpwd=${adminpassword} epicsettings=${epicsettings} externalport=${port} homeserver=${homeserver} homekingdom=${homekingdom} loginserver=${loginserver} maxplayers=${maxplayers} queryprort=${queryport} rmiport=${rmiport} rmiregport=${rmiregport} servername=${servername} serverpassword=${serverpassword} start=${start} ip=${ip}"
 
 #### LinuxGSM Settings ####
 

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -52,7 +52,7 @@ queryport="27016"
 
 ## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
 # Parameters are edited in the game server config file.
-startparameters="start='${gamemode}' adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryprort='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' start='${start}' ip='${ip}'"
+startparameters="start=${gamemode} adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryprort='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' start='${start}' ip='${ip}'"
 
 #### LinuxGSM Settings ####
 

--- a/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
+++ b/lgsm/config-default/config-lgsm/wurmserver/_default.cfg
@@ -39,6 +39,7 @@ homeserver="true"
 # connected with another server that is going to act as a loginserver.
 loginserver="true"
 
+# Used for server to server communication.
 rmiport="7221"
 rmiregport="7220"
 
@@ -51,7 +52,7 @@ queryport="27016"
 
 ## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
 # Parameters are edited in the game server config file.
-startparameters="start=${gamemode} adminpwd=${adminpassword} epicsettings=${epicsettings} externalport=${port} homeserver=${homeserver} homekingdom=${homekingdom} loginserver=${loginserver} maxplayers=${maxplayers} queryprort=${queryport} rmiport=${rmiport} rmiregport=${rmiregport} servername=${servername} serverpassword=${serverpassword} start=${start} ip=${ip}"
+startparameters="start='${gamemode}' adminpwd='${adminpassword}' epicsettings='${epicsettings}' externalport='${port}' homeserver='${homeserver}' homekingdom='${homekingdom}' loginserver='${loginserver}' maxplayers='${maxplayers}' queryprort='${queryport}' rmiport='${rmiport}' rmiregport='${rmiregport}' servername='${servername}' serverpassword='${serverpassword}' start='${start}' ip='${ip}'"
 
 #### LinuxGSM Settings ####
 

--- a/lgsm/modules/fix_wurm.sh
+++ b/lgsm/modules/fix_wurm.sh
@@ -10,14 +10,15 @@ moduleselfname="$(basename "$(readlink -f "${BASH_SOURCE[0]}")")"
 # First run requires start with no parms.
 # After first run new dirs are created.
 if [ ! -d "${serverfiles}/Creative" ]; then
-	parmsbypass=1
-	fixbypass=1
-	exitbypass=1
-	command_start.sh
-	fn_firstcommand_reset
-	sleep 10
-	exitbypass=1
-	command_stop.sh
-	fn_firstcommand_reset
-	unset parmsbypass
+	fixname="Copy Creative directory"
+	fn_fix_msg_start
+	cp -R "${serverfiles}/dist/Creative" "${serverfiles}/Creative"
+	fn_fix_msg_end
+fi
+
+if [ ! -d "${serverfiles}/Adventure" ]; then
+	fixname="Copy Adventure directory"
+	fn_fix_msg_start
+	cp -R "${serverfiles}/dist/Adventure" "${serverfiles}/Adventure"
+	fn_fix_msg_end
 fi

--- a/lgsm/modules/info_game.sh
+++ b/lgsm/modules/info_game.sh
@@ -2175,36 +2175,25 @@ fn_info_game_wmc() {
 	fi
 }
 
-# Config Type: custom (key-value)
-# Comment: #
-# Example: SERVERNAME=SERVERNAME
+# Config Type: parameters
+# Parameters: true
+# Comment:
+# Example: hostname='SERVERNAME'
+# Filetype: parameters
 fn_info_game_wurm() {
-	# Config
-	if [ ! -f "${servercfgfullpath}" ]; then
-		port="${zero}"
-		queryport="${zero}"
-		rconpassword="${unavailable}"
-		servername="${unavailable}"
-		serverpassword="${unavailable}"
-		adminpassword="${unavailable}"
-		maxplayers="${zero}"
-	else
-		port=$(grep "EXTERNALPORT=" "${servercfgfullpath}" | grep -v "#" | tr -cd '[:digit:]')
-		queryport=$(grep "QUERYPORT=" "${servercfgfullpath}" | grep -v "#" | tr -cd '[:digit:]')
-		servername=$(grep "SERVERNAME=" "${servercfgfullpath}" | sed -e 's/^[ \t]*//g' -e '/^\//d' -e 's/SERVERNAME//g' | tr -d '=\";,:' | sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
-		serverpassword=$(grep "SERVERPASSWORD=" "${servercfgfullpath}" | sed -e 's/^[ \t]*//g' -e '/^\//d' -e 's/SERVERPASSWORD//g' | tr -d '=\";,:' | sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
-		adminpassword=$(grep "ADMINPWD=" "${servercfgfullpath}" | sed -e 's/^[ \t]*//g' -e '/^\//d' -e 's/ADMINPWD//g' | tr -d '=\";,:' | sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
-		maxplayers=$(grep "MAXPLAYERS=" "${servercfgfullpath}" | grep -v "#" | tr -cd '[:digit:]')
-		configip=$(grep "IP" "${servercfgfullpath}" | sed -e 's/^[ \t]*//g' -e '/^#/d' -e 's/IP//g' | tr -d '=\";,:' | sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
-
-		# Not set
-		port="${port:-"3724"}"
-		queryport="${queryport:-"27017"}"
-		servername="${servername:-"NOT SET"}"
-		serverpassword="${serverpassword:-"NOT SET"}"
-		adminpassword="${adminpassword:-"NOT SET"}"
-		maxplayers="${maxplayers:-"0"}"
-	fi
+	gamemode="${gamemode:-"NOT SET"}"
+	homekingdom="${homekingdom:-"0"}"
+	adminpassword="${adminpassword:-"0"}"
+	epicsettings="${epicsettings:-"NOT SET"}"
+	homeserver="${homeserver:-"NOT SET"}"
+	loginserver="${loginserver:-"NOT SET"}"
+	rmiport="${rmiport:-"0"}"
+	rmiregport="${rmiregport:-"0"}"
+	maxplayers="${maxplayers:-"0"}"
+	servername="${servername:-"NOT SET"}"
+	# serverpassword="${serverpassword:-"NOT SET"}"
+	port="${port:-"0"}"
+	queryport="${queryport:-"0"}"
 }
 
 unavailable="${red}UNAVAILABLE${default}"

--- a/lgsm/modules/info_game.sh
+++ b/lgsm/modules/info_game.sh
@@ -2183,7 +2183,7 @@ fn_info_game_wmc() {
 fn_info_game_wurm() {
 	gamemode="${gamemode:-"NOT SET"}"
 	homekingdom="${homekingdom:-"0"}"
-	adminpassword="${adminpassword:-"0"}"
+	adminpassword="${adminpassword:-"NOT SET"}"
 	epicsettings="${epicsettings:-"NOT SET"}"
 	homeserver="${homeserver:-"NOT SET"}"
 	loginserver="${loginserver:-"NOT SET"}"

--- a/lgsm/modules/info_messages.sh
+++ b/lgsm/modules/info_messages.sh
@@ -1666,6 +1666,8 @@ fn_info_message_wurm() {
 		fn_port "header"
 		fn_port "Game" port tcp
 		fn_port "Query" queryport udp
+		fn_port "RMI" rmiport tcp
+		fn_port "RMI Registry" rmiregport tcp
 	} | column -s $'\t' -t
 }
 

